server/storage/mvcc/kvstore_txn.go

```diff
@@ -120,9 +126,20 @@ func (tr *storeTxnRead) rangeKeys(ctx context.Context, key, end []byte, curRev i
 				zap.Error(err),
 			)
 		}
+		totalBytes += int64(len(vs[0]))
+		count += 1
+		if ro.MaxBytes > 0 && totalBytes >= ro.MaxBytes {
+			break
+		}
+	}
+	if count != len(kvs) {
+		newkvs := make([]mvccpb.KeyValue, count)
+		copy(newkvs, kvs)
```

Comment:
Why do yo copy the kvs?

Reply:
We need to shrink the slice to the actual length. The RangeResult returned will be used in caller. For example, the length will be used to determine if there is 'More'

Reply:
No need to copy the slice, you just need to generate a new slice based something like below,
```
kvs = kvs[:count]
```

Reply:
Done. Thanks!